### ISSUE TITLE
Emulate Development with lambda runtime interface emulator

### ISSUE THREAD

**Issue by jpaulolins** [aws-samples/aws-lambda-adapter#19]
Hi Guys,

I'm trying to emulate lambda in a local development environment using lambda runtime interface emulator[1]. When making the invocation to lambda runtime, the adapter attempts to find something on /var/task instead of passing the event request to my HTTP daemon running at the same container. 

I tried to create /var/task/ into the container. After that, the adapter tried to find a bootstrap script without success. 

I started my daemon and runtime emulator using bash script in the ENTRYPOINT container. 

Any idea how I can use lambda runtime to test lambda in a development environment without deployment?

Regards.

[1] - https://github.com/aws/aws-lambda-runtime-interface-emulator

**Comment by bnusunny** [comment]
You don’t need the emulator. Just run your http server locally, test and debug using existing tools normally. When you need to delay to lambda, add the adapter to your container as extension. That’s all you need to do.


在 2022年2月23日，上午11:25，João Paulo Lins ***@***.***> 写道：

﻿

Hi Guys,

I'm trying to emulate lambda in a local development environment using lambda runtime interface emulator[1]. When making the invocation to lambda runtime, the adapter attempts to find something on /var/task instead of passing the event request to my HTTP daemon running at the same container.

I tried to create /var/task/ into the container. After that, the adapter tried to find a bootstrap script without success.

I started my daemon and runtime emulator using bash script in the ENTRYPOINT container.

Any idea how I can use lambda runtime to test lambda in a development environment without deployment?

Regards.

[1] - https://github.com/aws/aws-lambda-runtime-interface-emulator

**Comment by jpaulolins** [comment]
Hi Harold,

Thank you so much for your response.

I understand that the adapter could work transparently, but for our pipeline, we would have the opportunity to do some tests on the local environment without making Deploy. 

Today we use serverlessish, and we want to migrate to lambda-adapter because the code is well sustained and robust.  

We want to maintain our pipeline from development to production with local tests through the emulator for simulating some events coming from APIGW or ALB, for example.

Below I'll share my docker file, entry-point file, and one main.go file with a simple http-server to understand how we are working today. 

I'll be very grateful if you could suggest or advise how I can use the emulator with lambda-adapter.  

Tks.


Dockerfile
```
FROM golang:1.15 AS build
WORKDIR /teste
COPY . .
RUN CGO_ENABLED=0 go build -ldflags="-s -w" -o main

#FROM public.ecr.aws/c2t6n2x5/serverlessish:2 AS s

COPY --from=public.ecr.aws/awsguru/aws-lambda-adapter:0.2.0 /opt/extensions/lambda-adapter /opt/extensions/lambda-adapter

FROM alpine

RUN apk add --no-cache bash

ADD https://github.com/aws/aws-lambda-runtime-interface-emulator/releases/latest/download/aws-lambda-rie /usr/bin/aws-lambda-rie

RUN chmod 755 /usr/bin/aws-lambda-rie
COPY ./entry.sh /
RUN chmod 755 /entry.sh

COPY --from=build /teste/main /main

ENV PORT=8081 

ENTRYPOINT ["/entry.sh"]

```

entry.sh
```
#!/bin/sh

# Start the first process
./main &
status=$?
if [ $status -ne 0 ]; then
  echo "Failed to start my server: $status"
  exit $status
fi


if [ -z "${AWS_LAMBDA_RUNTIME_API}" ]; then
  exec /usr/bin/aws-lambda-rie "$@"
else
  exec "$@"
fi 

```

maing.go
```
package main

import (
	"net/http"
	"net/http/httputil"
)

func main() {
	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
		dump, _ := httputil.DumpRequest(r, true)
		w.Write(append([]byte("this is what i received on port 8081:\n\n"), dump...))
	})

	http.HandleFunc("/ping", func(w http.ResponseWriter, r *http.Request) {
		w.Write([]byte(`ok`))
	})

	err := http.ListenAndServe(":8081", nil)
	if err != nil {
		panic(err)
	}
}
```

**Comment by bnusunny** [comment]
I see. Let me check how to do this.

**Comment by jpaulolins** [comment]
Great! Thank you very much! I'll do new test as soon you release adjustments.

**Comment by jpaulolins** [comment]
Great! Thank you very much! I'll do new test as soon you release adjustments.

**Comment by bnusunny** [comment]
It's not a simple update. Rust Runtime 0.5.0 has major breaking changes. I'm working on it. Hopefully to get the new release out soon.

**Comment by bnusunny** [comment]
@jpaulolins I just released Adapter v0.3.0. Could you check if the new version solve this issue?

**Comment by bnusunny** [comment]
@jpaulolins I will close this issue for now. If you still see issues, please reopen this one.
